File: backend/src/main/java/events/Event.java
package events;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import java.time;
import java.utils;
import java.lang.String;

@Entity
public class Event {
  @Id
  private long id;

  private String title;
  private String description;
  private String image;
  private String location;
  private LocalDateTime date;
  private String type;
  private String link;
  private List<String> eventPhotos;
}

File: src/pages/events/event.ts
export class UpcomingEvent {
    title: string;
    description: string;
    image: string;
    date: Date;
    time: Date;
    location: string;
    type: string[];
    link: string;

    constructor(title: string, description: string, image: string, date: Date, time: Date, location: string, type: string[], link: string){
        this.title = title;
        this.description = description;
        this.image = image;
        this.date = date;
        this.time = time;
        this.location = location;
        this.type = type;
        this.link = link;
    }
}